generated/src/aws-cpp-sdk-bedrock-runtime/source/BedrockRuntimeClient.cpp

```diff
@@ -46,111 +47,150 @@ using ResolveEndpointOutcome = Aws::Endpoint::ResolveEndpointOutcome;
 
 namespace Aws {
 namespace BedrockRuntime {
-const char SERVICE_NAME[] = "bedrock";
 const char ALLOCATION_TAG[] = "BedrockRuntimeClient";
+const char SERVICE_NAME[] = "bedrock";
 }  // namespace BedrockRuntime
 }  // namespace Aws
 const char* BedrockRuntimeClient::GetServiceName() { return SERVICE_NAME; }
 const char* BedrockRuntimeClient::GetAllocationTag() { return ALLOCATION_TAG; }
 
+BedrockRuntimeClient::BedrockRuntimeClient(const Aws::Auth::BearerTokenAuthSignerProvider& bearerTokenProvider,
+                                           std::shared_ptr<BedrockRuntimeEndpointProviderBase> endpointProvider,
+                                           const BedrockRuntime::BedrockRuntimeClientConfiguration& clientConfiguration)
+    : AwsSmithyClientT(
+          clientConfiguration, GetServiceName(), "Bedrock Runtime", Aws::Http::CreateHttpClient(clientConfiguration),
+          Aws::MakeShared<BedrockRuntimeErrorMarshaller>(ALLOCATION_TAG),
+          endpointProvider ? endpointProvider : Aws::MakeShared<BedrockRuntimeEndpointProvider>(ALLOCATION_TAG),
+          Aws::MakeShared<smithy::GenericAuthSchemeResolver<>>(
+              ALLOCATION_TAG, Aws::Vector<smithy::AuthSchemeOption>({smithy::SigV4AuthSchemeOption::sigV4AuthSchemeOption,
+                                                                     smithy::BearerTokenAuthSchemeOption::bearerTokenAuthSchemeOption})),
+          {
```

Comment:
OK this is sort of what i was alluding to earlier should we just allow a user to override the whole map and say "bring whatever you want", so that we arent bearer token specific, just, if you want to override this mapping, bring your own mapping.